pkg/operator/ceph/object/user/controller_test.go

```diff
@@ -353,3 +357,127 @@ func TestBuildUpdateStatusInfo(t *testing.T) {
+					return &http.Response{
+						StatusCode: 200,
+						Body:       ioutil.NopCloser(bytes.NewReader([]byte(userCreateJSON))),
+					}, nil
+				}
+			}
+
+			return nil, fmt.Errorf("unexpected request: %q. method %q. path %q", req.URL.RawQuery, req.Method, req.URL.Path)
+		},
+	}
+	adminClient, err := admin.New("rook-ceph-rgw-my-store.mycluster.svc", "53S6B9S809NUP19IJ2K3", "1bXPegzsGClvoGAiJdHQD1uOW2sQBLAZM9j9VtXR", mockClient)
+	assert.NoError(t, err)
+	userConfig := generateUserConfig(objectUser)
+	r := &ReconcileObjectStoreUser{
+		objContext: &cephobject.AdminOpsContext{
+			AdminOpsClient: adminClient,
+		},
+		userConfig: &userConfig,
+	}
+
+	t.Run("Test 1: user with empty name", func(t *testing.T) {
+		err = r.createorUpdateCephUser(objectUser)
+		assert.Error(t, err)
+	})
+
+	t.Run("Test 2: user without any Quotas or Capabilities", func(t *testing.T) {
+		objectUser.Name = name
+		userConfig = generateUserConfig(objectUser)
+		r.userConfig = &userConfig
+		err = r.createorUpdateCephUser(objectUser)
+		assert.NoError(t, err)
+	})
+
+	t.Run("Test 3: setting MaxBuckets for the user", func(t *testing.T) {
+		objectUser.Spec.Quotas = &cephv1.ObjectUserQuotaSpec{MaxBuckets: &maxbucket}
+		userConfig = generateUserConfig(objectUser)
+		r.userConfig = &userConfig
+		err = r.createorUpdateCephUser(objectUser)
+		assert.NoError(t, err)
+	})
+
+	t.Run("Test 4: setting Capabilities for the user", func(t *testing.T) {
+		objectUser.Spec.Quotas = nil
+		objectUser.Spec.Capabilities = &cephv1.ObjectUserCapSpec{
+			User:   "read",
+			Bucket: "read",
+		}
+		userConfig = generateUserConfig(objectUser)
+		r.userConfig = &userConfig
+		err = r.createorUpdateCephUser(objectUser)
+		assert.NoError(t, err)
+	})
+
+	t.Run("Test 5: setting MaxObjects for the user", func(t *testing.T) {
+		objectUser.Spec.Capabilities = nil
+		objectUser.Spec.Quotas = &cephv1.ObjectUserQuotaSpec{MaxObjects: &maxobject}
+		userConfig = generateUserConfig(objectUser)
+		r.userConfig = &userConfig
+		err = r.createorUpdateCephUser(objectUser)
+		assert.NoError(t, err)
+	})
+
+	t.Run("Test 6: setting MaxSize for the user", func(t *testing.T) {
+		maxsize, err := resource.ParseQuantity(maxsizestr)
+		assert.NoError(t, err)
+		objectUser.Spec.Quotas = &cephv1.ObjectUserQuotaSpec{MaxSize: &maxsize}
+		userConfig = generateUserConfig(objectUser)
+		r.userConfig = &userConfig
+		err = r.createorUpdateCephUser(objectUser)
+		assert.NoError(t, err)
+	})
+
+	t.Run("Test 7: setting both Quotas and Capabilities for the user", func(t *testing.T) {
+		objectUser.Spec.Capabilities = &cephv1.ObjectUserCapSpec{
+			User:   "read",
+			Bucket: "read",
+		}
+		objectUser.Spec.Quotas.MaxObjects = &maxobject
+		objectUser.Spec.Quotas.MaxBuckets = &maxbucket
+		userConfig = generateUserConfig(objectUser)
+		r.userConfig = &userConfig
+		err = r.createorUpdateCephUser(objectUser)
+		assert.NoError(t, err)
+	})
+}
```

Comment:
Need test cases for the new ability to update/delete quotas. I think we should test updating all quotas, updating one quota, updating a quota and removing another, removing all quotas while `spec.Quotas != nil`, and removing all quotas while `spec.Quotas == nil`.